arroyo-controller/src/states/scheduling.rs

```diff
@@ -350,9 +350,16 @@ impl State for Scheduling {
             needs_commits,
         }) = checkpoint_info.clone()
         {
-            let mut metadata = StateBackend::load_checkpoint_metadata(&ctx.config.id, epoch)
+            let mut metadata = match StateBackend::load_checkpoint_metadata(&ctx.config.id, epoch)
                 .await
-                .unwrap_or_else(|| panic!("epoch {} not found for job {}", epoch, ctx.config.id));
+            {
+                Some(metadata) => metadata,
+                None => {
+                    let message = format!("epoch {} not found for job {}", epoch, ctx.config.id);
+                    return Err(fatal(message.clone(), anyhow!(message)));
```

Comment:
The message (first argument to fatal) is what's shown to the user in the web ui, so it should be short and easy for end users to understand. It also doesn't need to include the job id, because that's logged separately.

Reply:
True, I've changed it.
![image](https://github.com/ArroyoSystems/arroyo/assets/8881183/0ca30f6d-fcb6-4235-9109-30b54ea3bef6)